client/app/src/main/java/com/example/criengine/Adapters/NotificationAdapter.java

```diff
@@ -0,0 +1,80 @@
+package com.example.criengine.Adapters;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.example.criengine.Objects.Notification;
+import com.example.criengine.Objects.Profile;
+import com.example.criengine.R;
+import java.util.ArrayList;
+
+/*
+ * Custom ArrayAdapter that can be used to show notification instances in a ListView
+ * @version 1.0
+ */
+public class NotificationAdapter extends ArrayAdapter<Notification> {
+    private ArrayList<Notification> notificationItems;
+    private Context context;
+    private Profile profile;
+
+    /**
+     * Constructor. Extends off of the array adapter.
+     * @param context The context of the activity.
+     * @param notificationItems The ArrayList of notifications.
+     */
+    public NotificationAdapter(@NonNull Context context, @NonNull ArrayList<Notification> notificationItems, Profile profile) {
+        super(context, 0, notificationItems);
+        this.context = context;
+        this.notificationItems = notificationItems;
+        this.profile = profile;
+    }
+
+    @NonNull
+    @Override
+    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+        // Get Item Data
+        View view = convertView;
+
+        if( view == null ) {
+            LayoutInflater inflater = LayoutInflater.from(context);
+            view = inflater.inflate(R.layout.book_item, parent, false);
+        }
+
+        // Assign all view objects.
+        TextView notificationDescription = view.findViewById(R.id.bookNameTextView);
+        TextView notificationDate = view.findViewById(R.id.bookStatusTextView);
+        Button dismissButton = view.findViewById(R.id.bookActionButton);
+
+        // Get the object that was clicked on from the list.
```

Comment:
Get the current notification
or 
Get the notification in this position

Reply:
I'm not sure what this has to do with clicked

Reply:
oh lul u rite